hassio/src/addon-view/config/hassio-addon-config.ts

```diff
@@ -164,6 +165,20 @@ class HassioAddonConfig extends LitElement {
         err.body?.message || err
       }`;
     }
+    if (this.addon?.state === "started") {
+      const confirmed = await showConfirmationDialog(this, {
+        title: this.addon.name,
+        text: "Do you want to restart the add-on with your changes?",
+        confirmText: "restart add-on",
+      });
+      if (confirmed) {
+        try {
+          await restartHassioAddon(this.hass, this.addon.slug);
+        } catch (err) {
+          this._error = `Failed to restart addon, ${err.body?.message || err}`;
+        }
+      }
+    }
```

Comment:
Create a helper for this instead of copy pasting it :-)

Reply:
Better now? 😄